### PR TITLE
[staging] perl: 5.32.1 -> 5.34.0

### PR DESCRIPTION
**Pull request by stigtsp**
###### Motivation for this change

- Updates `perl` to 5.34.0, removes 5.30.3 and obsolete patches
   https://metacpan.org/release/XSAWYERX/perl-5.34.0/changes
- Sets perl 5.34.0 as default in `unit`

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stigtsp** [comment]
@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.perl534
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.perl534
@GrahamcOfBorg build pkgsMusl.perl534
@GrahamcOfBorg build pkgsi686Linux.perl534
@GrahamcOfBorg build perl534

@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.perlPackages.HTTPDaemon
@GrahamcOfBorg build perlPackages.DBDmysql perlPackages.IOSocketSSL
@GrahamcOfBorg build hydra-unstable

**Comment by aanderse** [comment]
Thanks @stigtsp ❤️

File: pkgs/development/interpreters/perl/default.nix [hunk 1]
```diff
@@ -37,15 +37,11 @@ let
     patches =
       [
         # Do not look in /usr etc. for dependencies.
-        (if (versionOlder version "5.31.1") then ./no-sys-dirs-5.29.patch
```
Comment (zakame) [inline pkgs/development/interpreters/perl/default.nix:40]: Should `no-sys-dirs-5.29.patch` be deleted as well?

File: pkgs/development/interpreters/perl/default.nix [hunk 2]
```diff
@@ -37,15 +37,11 @@ let
...
-         else ./no-sys-dirs-5.31.patch)
+        ./no-sys-dirs-5.31.patch
       ]
       ++ optional stdenv.isSunOS ./ld-shared.patch
       ++ optionals stdenv.isDarwin [ ./cpp-precomp.patch ./sw_vers.patch ]
-      ++ optional crossCompiling ./MakeMaker-cross.patch
-      # Backporting https://github.com/Perl/perl5/pull/17946, can be
-      # removed if there's ever a 5.30.x release with it included.
-      ++ optional (versionOlder version "5.32.1") ./aarch64-darwin.patch;
```
Comment (zakame) [inline pkgs/development/interpreters/perl/default.nix:48]: Can we also drop `aarch64-darwin.patch` now too?

**Review by zakame — APPROVED** [review]
LGTM, thanks!  Just a couple minor queries I think...

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -18932,14 +18932,14 @@ in
   ### DEVELOPMENT / PERL MODULES
 
   perlInterpreters = callPackages ../development/interpreters/perl {};
-  inherit (perlInterpreters) perl530 perl532 perldevel;
+  inherit (perlInterpreters) perl532 perl534 perldevel;
 
-  perl530Packages = recurseIntoAttrs perl530.pkgs;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:18937]: Do we want to add an alias for the removed version?

File: pkgs/development/interpreters/perl/default.nix (hunk 1 above)
Comment (stigtsp) [reply pkgs/development/interpreters/perl/default.nix:40]: Thx! Forgot to remove the files as well. Fixed.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (stigtsp) [reply pkgs/top-level/all-packages.nix:18937]: Hm, I don't think so. If someone uses `perl530` rather than `perl` they would expect to get 5.30.x?

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perl534 perl532 perl
@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.perlPackages.HTTPDaemon

**Comment by stigtsp** [comment]
@GrahamcOfBorg build unit

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (zakame) [reply pkgs/top-level/all-packages.nix:18937]: Yep, and we already remove the `withPerl530` predicate above so an alias would probably end up undefined.

This also makes it more consistent with https://metacpan.org/pod/perlpolicy#MAINTENANCE-AND-SUPPORT (supporting only the last 2 Perl stable releases.)

**Review by Izorkin — APPROVED** [review]

**Comment by mweinelt** [comment]
Thanks, I just build perl534 locally and it looks fine.

**Pull request merged**